Content.Server/GameObjects/Components/Gravity/GravityGeneratorComponent.cs

```diff
@@ -98,29 +98,25 @@ bool IAttackHand.AttackHand(AttackHandEventArgs eventArgs)
             return true;
         }
 
-        public bool AttackBy(AttackByEventArgs eventArgs)
+        public bool WelderAct(WelderActEventArgs eventArgs)
         {
-            if (!eventArgs.AttackWith.TryGetComponent<WelderComponent>(out var welder)) return false;
-            if (welder.TryUse(5.0f))
-            {
-                // Repair generator
-                var damagable = Owner.GetComponent<DamageableComponent>();
-                var breakable = Owner.GetComponent<BreakableComponent>();
-                damagable.HealAllDamage();
-                breakable.broken = false;
-                _intact = true;
+            var welder = (WelderComponent)eventArgs.ToolComponent;
```

Comment:
If there should always be a `WelderComponent` doing the welding then the `eventArgs` should expose that type directly. If this is not the case then this code is buggy.

Reply:
This isn't an issue anymore.